src/hotspot/share/prims/jvmtiEnvBase.cpp

```diff
@@ -1514,6 +1515,13 @@ JvmtiEnvBase::get_object_monitor_usage(JavaThread* calling_thread, jobject objec
     // Get the actual set of threads trying to enter, or re-enter, the monitor.
     wantList = Threads::get_pending_threads(tlh.list(), nWant + nWait, (address)mon);
     nWant = wantList->length();
+    for(jint i = 0; i < nWant; i++) {
+      JavaThread* w = wantList->at(i);
+      oop thread_oop = get_vthread_or_thread_oop(w);
+      if (thread_oop->is_a(vmClasses::BaseVirtualThread_klass())) {
+        nWant_Skip++;
+      }
+    }
```

Comment:
Thank you for taking care about this issue.
The nWant_Skip and and the fragment with lines 1518-1524 would not be needed if the function `Threads::get_pending_threads()` is fixed instead:
```
-     if (java_lang_VirtualThread::is_instance(thread_oop)) {
+    if (thread_oop->is_a(vmClasses::BaseVirtualThread_klass())) {
``

Reply:
Thank you for your correction. The code has been updated accordingly your suggest. I need redo the additional testing to verify it.